src/api/api.ts

```diff
@@ -0,0 +1,19 @@
+const API_URL = 'https://jsonplaceholder.typicode.com/';
+
+export const getData = async <T>(url: string): Promise<T> => {
+  const response = await fetch(url);
+
+  return response.json();
+};
+
+export const getPosts = (): Promise<Posts> => {
+  return getData(`${API_URL}posts`);
+};
+
+export const getUsers = (): Promise<Users> => {
+  return getData(`${API_URL}users`);
+};
+
+export const getComments = (): Promise<Comments> => {
+  return getData(`${API_URL}comments`);
+};
```

Comment:
в функции getData ты ожидаешь T, но не передаешь
```suggestion
export const getData = async <T>(url: string): Promise<T> => {
  const response = await fetch(url);

  return response.json();
};

export const getPosts = () => {
  return getData<Posts>(`${API_URL}posts`);
};

export const getUsers = () => {
  return getData<Users>(`${API_URL}users`);
};

export const getComments = () => {
  return getData<Comments>(`${API_URL}comments`);
};
```
код который ты пишешь нужно понимать, а не просто повторять)

Reply:
Согласен, не до конца был разобрался, что какая функция принимает и возвращает